e2e/internal/kubeclient/deploy.go

```diff
@@ -220,36 +277,23 @@ retryLoop:
 				}
 				return origErr
 			}
-			switch evt.Type {
-			case watch.Added:
-				fallthrough
-			case watch.Modified:
-				pods, err := resource.getPods(ctx, c, namespace, name)
+			switch condition {
+			case Ready:
+				ready, err := c.checkIfReady(ctx, name, namespace, evt, resource)
 				if err != nil {
 					return err
 				}
-				numPodsReady := 0
-				for _, pod := range pods {
-					if isPodReady(&pod) {
-						numPodsReady++
-					}
+				if ready {
+					return nil
 				}
-				desiredPods, err := resource.numDesiredPods(evt.Object)
+			case InitContainersRunning:
+				running, err := c.checkIfRunning(ctx, name, namespace, resource)
 				if err != nil {
 					return err
 				}
-				if desiredPods <= numPodsReady {
-					// Wait for 5 more seconds just to be *really* sure that
-					// the pods are actually up.
-					sleep, cancel := context.WithTimeout(ctx, time.Second*5)
-					defer cancel()
-					<-sleep.Done()
```

Comment:
This PR silently removes this sleep, since CI is happy with that, I'm okay with that.